home/tests/test_api.py

```diff
@@ -2,234 +2,244 @@
 import queue
 from pathlib import Path
 
-from django.contrib.auth import get_user_model
-from django.test import Client, TestCase
+import pytest
 from wagtail import blocks
 
 from home.content_import_export import import_content
-from home.tests.page_builder import MBlk, MBody, PageBuilder, WABlk, WABody
-
-from .utils import create_page
-
-from home.models import (  # isort:skip
+from home.models import (
     ContentPage,
     HomePage,
     OrderedContentSet,
     PageView,
     VariationBlock,
 )
 
+from .page_builder import MBlk, MBody, PageBuilder, WABlk, WABody
+from .utils import create_page
 
-class PaginationTestCase(TestCase):
-    @classmethod
-    def setUpTestData(self):
-        home_page = HomePage.objects.first()
-        main_menu = PageBuilder.build_cpi(home_page, "main-menu", "Main Menu")
-        content_page1 = PageBuilder.build_cp(
-            parent=main_menu,
-            slug="main-menu-first-time-user",
-            title="main menu first time user",
-            bodies=[
-                WABody(
-                    "main menu first time user", [WABlk("*Welcome to HealthAlert* 🌍")]
-                ),
-                MBody(
-                    "main menu first time user", [MBlk("*Welcome to HealthAlert* 🌍")]
-                ),
-            ],
-            tags=["menu"],
-            quick_replies=["Self-help", "Settings", "Health Info"],
-            triggers=["Main menu"],
-        )
-        PageBuilder.build_cp(
-            parent=content_page1,
-            slug="health-info",
-            title="health info",
-            bodies=[
-                WABody("health info", [WABlk("*Health information* 🏥")]),
-                MBody("health info", [MBlk("*Health information* 🏥")]),
-            ],
-            tags=["health_info"],
-        )
-        PageBuilder.build_cp(
-            parent=content_page1,
-            slug="self-help",
-            title="self-help",
-            bodies=[
-                WABody("self-help", [WABlk("*Self-help programs* 🌬️")]),
-                MBody("self-help", [MBlk("*Self-help programs* 🌬️")]),
-            ],
-            tags=["self_help"],
-        )
-
-    def setUp(self):
-        self.user_credentials = {"username": "test", "password": "test"}
-        self.user = get_user_model().objects.create_user(**self.user_credentials)
-        self.client.login(**self.user_credentials)
-        self.content_page1 = ContentPage.objects.first()
-        self.content_page2 = ContentPage.objects.last()
 
-    def test_login_required(self):
+@pytest.fixture()
+def uclient(client, django_user_model):
+    creds = {"username": "test", "password": "test"}
+    django_user_model.objects.create_user(**creds)
+    client.login(**creds)
+    return client
+
+
+@pytest.fixture()
```

Comment:
Do we need to set the scope of this fixture, to avoid it being recreated every test, or is the default scope fine?

Reply:
Each test function runs in its own db transaction (as with Django's `TestCase`), so we need to populate the db for every test. (We were doing that with the `setUp()` methods and such anyway, so there's no real change here.)

Reply:
The `setUpTestData` method did it once per class, and then ran all the tests against that setup state, I assume with transaction checkpoints or something.

Reply:
Ah, I forgot about that. It doesn't seem to make the tests any slower, so I don't think it's a problem.